packages/app/src/systems/DApp/machines/connectRequestMachine.ts

```diff
@@ -183,26 +193,39 @@ export const connectRequestMachine = createMachine(
           return undefined;
         },
       }),
-      addConnection: FetchMachine.create<Connection, Connection | undefined>({
+      addConnection: FetchMachine.create<
+        MachineContext,
+        Connection | undefined
+      >({
         showError: false,
         fetch: async ({ input }) => {
-          if (!input?.origin || !Array.isArray(input?.accounts)) {
-            throw new Error('Origin or account not passed');
+          if (
+            !input?.origin ||
+            !Array.isArray(input?.selectedAddresses) ||
+            input?.selectedAddresses.length === 0
+          ) {
+            throw new Error('Invalid connect request');
+          }
+          if (input.connection) {
+            return ConnectionService.updateConnectedAccounts({
+              origin: input.origin,
+              accounts: input.selectedAddresses,
+            });
           }
           return ConnectionService.addConnection({
             data: {
               origin: input.origin,
-              accounts: input.accounts,
-              favIconUrl: input.favIconUrl,
-              title: input.title,
+              accounts: input.selectedAddresses,
+              favIconUrl: input.favIconUrl!,
```

Comment:
@luizstacio we should find some way to remove the `!` right?  The easiest way is to add a check such as
```
if (!input.favIconUrl || !input.title) throw Error ("...");
```
is there a better way

Reply:
In some cases, it's ok to use it, but I think in this case is not required, it should accept undefined...